discord/app_commands/commands.py

```diff
@@ -2430,6 +2473,135 @@ def inner(f: T) -> T:
         return inner(func)
 
 
+def private_channel_only(func: Optional[T] = None) -> Union[T, Callable[[T], T]]:
```

Comment:
This should either drop the ability to optionally use parens or add the necessary overloads here otherwise you'll get type errors.